dev-tools/omdb/src/bin/omdb/nexus.rs

```diff
@@ -1048,3 +1074,102 @@ async fn cmd_nexus_sled_add(
     eprintln!("added sled {} ({})", args.serial, args.part);
     Ok(())
 }
+
+/// Runs `omdb nexus sleds expunge`
+async fn cmd_nexus_sled_expunge(
+    client: &nexus_client::Client,
+    args: &SledExpungeArgs,
+    omdb: &Omdb,
+    log: &slog::Logger,
+) -> Result<(), anyhow::Error> {
+    // This is an extremely dangerous and irreversible operation. We put a
+    // couple of safeguards in place to ensure this cannot be called without
+    // due consideration:
+    //
+    // 1. We'll require manual input on stdin to confirm the sled to be removed
+    // 2. We'll warn sternly if the sled-to-be-expunged is still present in the
+    //    most recent inventory collection
+    use nexus_db_model::Sled;
+    use nexus_db_queries::context::OpContext;
+
+    // First, we need to look up the sled so we know it's serial number.
+    let datastore = args.db_url_opts.connect(omdb, log).await?;
+    let opctx = OpContext::for_tests(log.clone(), datastore.clone());
+    let opctx = &opctx;
+
+    let sled = {
+        use async_bb8_diesel::AsyncRunQueryDsl;
+        use diesel::ExpressionMethods;
+        use diesel::QueryDsl;
+        use diesel::SelectableHelper;
+        use nexus_db_queries::db::schema::sled::dsl;
+
+        let conn = datastore.pool_connection_for_tests().await?;
+        dsl::sled
+            .filter(dsl::id.eq(args.sled_id))
+            .select(Sled::as_select())
+            .first_async::<Sled>(&*conn)
+            .await
+            .with_context(|| format!("failed to find sled {}", args.sled_id))?
```

Comment:
Can this use `lookup_path` to look up the sled?

Reply:
Yes, yes it can. I always forget that path.

Reply:
same 😅